treeherder/webapp/api/runnable_jobs.py

```diff
@@ -57,6 +78,70 @@ def list(self, request, project):
                 'state': 'runnable',
                 'result': 'runnable'})
 
+        for node in tc_graph['tasks']:
+            build_platform = node['task']['extra']['treeherder']['build']['platform']
+            try:
+                build_platform_data = models.BuildPlatform.objects.get(platform=build_platform)
+                build_platform_id = build_platform_data.id
+                build_os = build_platform_data.os_name
+                build_architecture = build_platform_data.architecture
+            except models.BuildPlatform.DoesNotExist:
+                build_platform_id = ""
+                build_os = ""
+                build_architecture = ""
+
+            machine_platform = node['task']['extra']['treeherder']['machine']['platform']
+            try:
+                machine_platform_data = models.MachinePlatform.objects.get(platform=machine_platform)
+                machine_platform_id = machine_platform_data.id
+                machine_platform_os = machine_platform_data.os_name
+                machine_platform_architecture = machine_platform_data.architecture
+            except models.MachinePlatform.DoesNotExist:
+                machine_platform_id = ""
+                machine_platform_os = ""
+                machine_platform_architecture = ""
+
+            job_type_name = node['task']['metadata']['name']
+            try:
+                job_type_data = models.JobType.objects.get(name=job_type_name)
+                job_type_id = job_type_data.id
+            except models.JobType.DoesNotExist:
+                job_type_id = ""
+
+            if 'groupName' in node['task']['extra']['treeherder']:
```

Comment:
this could be stated as:

    job_group_name = node['task']['extra']['treeherder'].get(groupName, "")

instead of the if...else